tsl/src/bgw_policy/compression_api.c

```diff
@@ -106,25 +106,20 @@ policy_compression_add(PG_FUNCTION_ARGS)
 	Oid compress_after_type = PG_ARGISNULL(1) ? InvalidOid : get_fn_expr_argtype(fcinfo->flinfo, 1);
 	bool if_not_exists = PG_GETARG_BOOL(2);
 	Interval *default_schedule_interval = DEFAULT_SCHEDULE_INTERVAL;
-
-	PreventCommandIfReadOnly("add_compression_policy()");
-
 	Hypertable *hypertable;
 	Cache *hcache;
 	Dimension *dim;
-	ts_hypertable_permissions_check(ht_oid, GetUserId());
-	Oid owner_id = ts_hypertable_permissions_check(ht_oid, GetUserId());
+	Oid owner_id;
+
+	PreventCommandIfReadOnly("add_compression_policy()");
```

Comment:
To avoid inconsistencies between the function name and this text:
```suggestion
	PreventCommandIfReadOnly(get_func_name(FC_FN_OID(fcinfo)));
```